packages/firestore/src/remote/remote_store.ts

```diff
@@ -139,8 +124,10 @@ export class RemoteStore {
     private localStore: LocalStore,
     /** The client-side proxy for interacting with the backend. */
     private datastore: Datastore,
-    private onlineStateHandler: (onlineState: OnlineState) => void
-  ) {}
+    onlineStateHandler: (onlineState: OnlineState) => void
+  ) {
+    this.onlineStateTracker = new OnlineStateTracker(onlineStateHandler);
```

Comment:
Might as well pass the `OnlineStateTracker` in so the RemoteStore doesn't even have to know about the onlineStateHandler.

Reply:
I don't think there's a strong case either way. The general principle I was following was to invert dependencies where possible, the general rule being to pass your dependencies rather than new them up.

This makes it possible to construct a testing OnlineStateTracker if we ever wanted to do so and pushes all the object construction into one place. While it's true that FirestoreClient doesn't need to know about OnlineStateTracker, FirestoreClient is the assembly point for the rest of our constellation of objects so it doesn't seem like a bad place.